### DISCUSSION TITLE
Duplicate Files video_2021-02-16_20-07-24_7143747620735547392.mp4

### DISCUSSION THREAD

**Discussion by atomictoyguy** [Dineshkarthik/telegram_media_downloader#330] [General]
So I have noticed on repeat runs of this script I often get duplicates of files with the naming conventions like video_2021-02-16_20-07-24_7143747620735547392.mp4, it does it with photos as well if they have that naming. 

I can tell that they are actually duplicates, and not just duplicates to the channel because the channel only has 2094 videos, and so far my video folder has 2619 videos in it after running the script probably 8 times or so. 

Note I am running the Rhillip patch. 

My instinct is that it is some quirk about how Telegram serves up the media, where it assigns a random hash to the end of the file or something. But I have no idea, I know the script is not supposed to be downloading duplicates if the file already exists. So I don't think it is the script causing it necessarily. Does anyone have any insight into this?

**Comment by tangyoha** [comment]
@atomictoyguy, you can first name the file with `message.id`, this action will help determine the unique file.
